juju/juju
# Remove checks about spaces on endpoint bindings

We currently make multiple checks when adding or updating endpoint bindings, specially we check that the bound space actually exists.

We get rid of these checks in order to be able to migrate to the new dqlite domain and add TODOs for when we migrate the endpoint bindings domain.

The state tests around these checks were simply removed. This will be rewritten when migrating endpoint bindings to dqlite.

_Note: these checks were mainly needed for backwards compatibility. This has been added to the risk register._

## Checklist


## QA steps

Binding should work as expected:

```
juju bootstrap localhost c
juju add-model m
juju add-space test-space 10.170.236.0/24 # use the correct subnet in your case
juju deploy influxdb
juju bind influxdb test-space
moving endpoint "admin" from space "alpha" to "test-space"
moving endpoint "grafana-source" from space "alpha" to "test-space"
moving endpoint "nrpe-external-master" from space "alpha" to "test-space"
moving endpoint "query" from space "alpha" to "test-space"
```

## Links

**Jira card:** JUJU-5739
